packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+      modelConfig: {
+        type: 'number'
+      }
+    }
+  },
+
+  computed: {
+    showCalendar () {
+      return (this.mode === 'custom')
+    }
+  },
+
+  watch: {
+    selectimedTimeRange: {
+      handler (newValue, oldValue) {
+        // console.warn('>>>> selectimedTimeRange watcher <<<')
+        // console.warn(newValue)
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    /**
+     * TODO: Given a relative time frame, determine `start` and `end` timestamps in UTC
+     * https://bloop.ai/search/how-to-use-date-fns-library-round-to-nearest-minutes-in-javascript
+     * @param {*} timeframe
+     */
+    changeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+    },
+    clearSelection () {
+      this.selectimedTimeRange = ''
+      this.selectedTimeframe = this.allowedTimePeriods[0]
+    },
+    submitTimeFrame () {
+      console.warn('>>> submitTimeFrame')
+      let start = 0
+      let end = 0
+
+      // If calendar currently shown, send start/end values
+      if (this.showCalendar) {
+        start = Math.floor(this.selectimedTimeRange.start / 1000)
+        end = Math.floor(this.selectimedTimeRange.end / 1000)
+      } else {
+        end = getUnixTime(roundToNearestMinutes(Date.now()))
+        start = end - this.selectedTimeframe.timeframeLength
+      }
+
+      console.log({ start, end })
+
+      this.$emit('changed', { start, end })
+    }
+  }
+}
+
+</script>
+
+<style lang="scss">
+@import '~@kongponents/styles/variables';
+
+$margin: .2rem;
+
+.analytics-timepicker {
+  // TODO: consider moving to a new button theme
+  .timepicker-input {
+    padding: var(--spacing-sm) var(--spacing-xl) var(--spacing-sm) 40px !important;
+    background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M14 14H2V2H4V4H6V2H10V4H12V2H14V14ZM14 0H2C0.895 0 0 0.896 0 2V14C0 15.104 0.895 16 2 16H14C15.104 16 16 15.104 16 14V2C16 0.896 15.104 0 14 0ZM7 12H9V10H7V12ZM10 9H12V7H10V9ZM7 9H9V7H7V9ZM4 9H6V7H4V9ZM4 12H6V10H4V12Z' fill='%236F7787'/%3E%3C/svg%3E%0A");
+    background-repeat: no-repeat;
+    background-position: 12px 50%;
+    color: var(--grey-600) !important;
+    font-weight: 500;
+
+    &:focus {
+      box-shadow: none !important;
+    }
+  }
+  .k-popover {
+    min-width: 22rem;
+
+    // v-calendar overrides
+    .vc-pane-container {
+      .vc-month, .vc-day {
+        color: var(--grey-500);
+      }
+
+      // AM / PM highlights
+      .vc-am-pm button.active {
+        background: var(--blue-500);
+      }
+
+      // Hide clock icon
+      .vc-time-icon {
+        display: none;
+      }
+      // TODO: current day highligh should be `blue-500` as well
+      // .vc-day-content.vc-focusable
+
+      // Time range highlight
+      .vc-day .vc-highlights.vc-day-layer .vc-highlight {
+        background-color: var(--blue-100);
+      }
```

Comment:
If these are all `v-calendar` overrides, would it make sense to put them in a separate block? This would allow us to keep them separate in case they change over time.

Reply:
yup! I've broken things out into two separate (unscoped) SCSS sections

Everything owned / created by us
```
.k-datetime-picker { 
  .timepicker-input { ... }
  .k-popover { ... }
```

V-calendar overrides
```
.k-datetime-picker {
  .vc-container { ... }
}
```